core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -5346,29 +5326,55 @@ public void setKeyRotationTime(@Nullable Date time) {
      * The given time cannot be in the future.
      * </p>
      */
-    public void setKeyRotationTime(long unixTimeSeconds) {
-        checkArgument(unixTimeSeconds <= TimeUtils.currentTimeSeconds(), "Given time (%s) cannot be in the future.",
-                TimeUtils.dateTimeFormat(unixTimeSeconds * 1000));
-        vKeyRotationTimestamp = unixTimeSeconds;
+    public void setKeyRotationTime(Instant keyRotationTime) {
```

Comment:
`Instant` should probably be annotated as `@Nullable`.

Also, The JavaDoc should be updated as it currently says:

> You can stop key rotation by calling this method again with {@code 0} as the argument.

When updating the JavaDoc it would be good to add an actual `@param` line for the argument.

Reply:
Both done.